src/xradio/vis/_vis_utils/_ms/partition_queries.py

```diff
@@ -14,6 +14,137 @@
 from .subtables import subt_rename_ids
 
 
+def enumerated_product(*args):
+    yield from zip(
+        itertools.product(*(range(len(x)) for x in args)), itertools.product(*args)
+    )
+
+
+def create_partitions(in_file: str, partition_scheme: Union[str, list], vla_otf=False):
+    """Create a list of dictionaries with the partition information.
+
+    Args:
+        in_file (str): Input MSv2 file path.
+        partition_scheme (Union[str, list]): Partition scheme to be used.
+        vla_otf (bool, optional):  The partioning of VLA OTF (on the fly) mosaics needs a special partitioning scheme. Defaults to False.
+
+    Returns:
+        list: list of dictionaries with the partition information.
+    """
+
+    # Create partition table
+    from casacore import tables
```

Comment:
Are these imports here for some reason? If not I'd move it up the module level when reorganizing this file.